krabs/krabs/filtering/event_filter.hpp

```diff
@@ -22,8 +22,10 @@ namespace krabs { namespace details {
 
 namespace krabs {
 
-    typedef void(*c_provider_callback)(const EVENT_RECORD &, const krabs::trace_context &);
```

Comment:
Is this a breaking change? Would it make more sense if we kept c_provider_callback and just added the c_provider_error_callback? 

We can make the change but that would be another argument for a major version bump

Reply:
Good point, I made this update to improve readability but we should prioritize back-compat. Reverted.